crates/full-node/sov-sequencer/src/preferred/block_executor.rs

```diff
@@ -144,27 +141,49 @@ impl<S: Spec, Rt: Runtime<S>> RollupBlockExecutor<S, Rt> {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
         info: &StateUpdateInfo<S::Storage>,
-        rollup_exec_config: RollupBlockExecutorConfig<S, Rt>,
+        rollup_exec_config: RollupBlockExecutorConfig<S>,
+        seq_config: SequencerConfig<S::Address, PreferredSequencerConfig>,
+    ) -> RollupBlockExecutor<S, Rt> {
+        Self::new_with_tx_cache_writer_inner(info, None, rollup_exec_config, seq_config)
+    }
+
+    pub fn new_with_tx_cache_writer(
+        info: &StateUpdateInfo<S::Storage>,
+        tx_cache_writer: TxResultWriter<S, Rt>,
+        rollup_exec_config: RollupBlockExecutorConfig<S>,
+        seq_config: SequencerConfig<S::Address, PreferredSequencerConfig>,
     ) -> RollupBlockExecutor<S, Rt> {
+        Self::new_with_tx_cache_writer_inner(
+            info,
+            Some(tx_cache_writer),
+            rollup_exec_config,
+            seq_config,
+        )
+    }
+
+    fn new_with_tx_cache_writer_inner(
```

Comment:
Also very tiny nit: maybe just `new_inner` is clearer here, since it can be also called without a tx cache writer?